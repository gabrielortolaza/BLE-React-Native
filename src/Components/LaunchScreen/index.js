import React, { PureComponent } from "react";
import {
  View,
} from "react-native";
import PropTypes from "prop-types";
import SplashScreen from "react-native-splash-screen";
import auth from "@react-native-firebase/auth";

import StyleSheet from "../../Proportional";
import { Colors } from "../../Themes";

export default class LaunchScreen extends PureComponent {
  componentDidMount() {
    const { navigation } = this.props;
    const user = auth().currentUser;
    setTimeout(() => {
      if (user) {
        navigation.navigate("Tabs");
      } else {
        navigation.navigate("TourStart");
      }
      SplashScreen.hide();
    }, 1500);
  }

  render() {
    return (
      <View
        testID="LaunchScreenContainer"
        style={styles.container}
      >
        {/* <Animatable.Image
          animation="zoomIn"
          duration={1600}
          iterationCount="infinite"
          useNativeDriver
          style={styles.logoImage}
          source={Images.splashSupergenie}
        >
        </Animatable.Image> */}
      </View>
    );
  }
}

LaunchScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

const styles = StyleSheet.createProportional({
  container: {
    flex: 1,
    backgroundColor: Colors.blue,
    alignItems: "center",
    justifyContent: "center"
  },
  logoImage: {
    width: "78%",
    resizeMode: "contain"
  }
});
