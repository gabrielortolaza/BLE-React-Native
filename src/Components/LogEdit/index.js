import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, Text } from "react-native";

import StyleSheet from "../../Proportional";
import { Fonts } from "../../Themes";
import { RoundButton, ConfirmationToast } from "../Shared";
import * as M from "../../Config/messages";

import RowDateTime from "../ModalSessionScreen/RowDateTime";
import RowVolumeNotes from "../ModalSessionScreen/RowVolumeNotes";

export default class Editing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      focus: "",
      error: false,
      shouldConfirm: false,
    };
  }

  onSave = () => {
    const { sessionSave, closeModal } = this.props;
    sessionSave()
      .then(closeModal)
      .catch((x) => this.setState({ error: x }));
  };

  onConfirmSave = () => {
    const { volume, sessionType } = this.props;
    if (sessionType === "pump") {
      if (volume === 0 || volume === null) {
        this.setState({ shouldConfirm: true });
      } else {
        this.onSave();
      }
    } else {
      this.onSave();
    }
  };

  onConfirmDeny = () => {
    this.setState({ shouldConfirm: false });
  };

  render() {
    const {
      onChange,
      startedAt,
      finishedAt,
      volume,
      sessionType,
      notes,
      closeModal,
      measureUnit,
    } = this.props;
    const { focus, shouldConfirm, error } = this.state;

    return (
      <View style={styles.container}>
        <View style={styles.popup}>
          <Text style={styles.title}>Edit Session</Text>

          <RowDateTime
            finishedAt={finishedAt || Date.now()}
            focus={focus}
            onChange={onChange}
            onFocus={(focusId) => this.setState({ focus: focus === focusId ? "" : focusId })}
            startedAt={startedAt}
          />

          <RowVolumeNotes
            focus={focus}
            hideVolume={sessionType === "feed"}
            measureUnit={measureUnit}
            notes={notes}
            onChange={onChange}
            onFocus={(focusId) => this.setState({ focus: focus === focusId ? "" : focusId })}
            volume={volume}
          />
          {error && <Text style={styles.error}>{error}</Text>}
          <View style={styles.buttons}>
            <RoundButton white small onPress={closeModal}>
              Cancel
            </RoundButton>
            <RoundButton small onPress={this.onConfirmSave}>
              Save
            </RoundButton>
          </View>

          {shouldConfirm && (
            <ConfirmationToast
              title={M.NO_AMOUNT}
              subtitle={M.CONTINUE_NO_AMOUNT}
              onPressConfirm={this.onSave}
              onPressDeny={this.onConfirmDeny}
            />
          )}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.createProportional({
  container: {
    flex: 1,
    padding: 30,
    backgroundColor: "rgba(212, 236, 246, 0.7)",
    justifyContent: "center",
    alignItems: "center",
  },
  popup: {
    paddingVertical: 30,
    paddingHorizontal: 25,
    backgroundColor: "white",
    borderRadius: 4,
  },
  title: {
    marginBottom: 25,
    textAlign: "center",
    ...Fonts.SemiBold,
    fontSize: 14,
  },
  error: {
    marginTop: 10,
    textAlign: "center",
    ...Fonts.SemiBold,
    fontSize: 12,
    color: "red",
  },
  buttons: {
    flexDirection: "row",
    marginTop: 25,
    justifyContent: "space-between",
  },
});

Editing.propTypes = {
  sessionSave: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,

  startedAt: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  finishedAt: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  volume: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  notes: PropTypes.string,
  sessionType: PropTypes.string,
  measureUnit: PropTypes.string,
};
