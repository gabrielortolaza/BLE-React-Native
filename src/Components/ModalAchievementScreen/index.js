/** Page appears to no longer be used */
/* eslint-disable */
// import React, { Component } from 'react'
// import {
//   View,
//   Image,
//   Animated,
//   Dimensions,
//   Text,
//   Modal
// } from 'react-native'
// import PropTypes from 'prop-types'
// import LinearGradient from 'react-native-linear-gradient'
// // import LottieView from 'lottie-react-native'

// import StyleSheet from '../../Proportional'
// import { RoundButton } from '../Shared/index'
// import Fonts from '../../Themes/Fonts'

// const Lighting1 = require('../../Assets/Animations/Lighting1.json')
// const Lighting2 = require('../../Assets/Animations/Lighting2.json')

// export const getAchievementImage = achievementId => {
//   switch (achievementId) {
//     case 'age_1':
//     case 'streak_1':
//     case 'volume_1':
//     case 'badge_1':
//       return require('../../Assets/Images/Milestones/step1.png')
//     case 'age_2':
//     case 'streak_2':
//     case 'volume_2':
//     case 'volume_3':
//     case 'badge_2':
//       return require('../../Assets/Images/Milestones/step2.png')
//     case 'age_3':
//     case 'badge_3':
//       return require('../../Assets/Images/Milestones/step3.png')
//     case 'age_4':
//     case 'badge_4':
//       return require('../../Assets/Images/Milestones/step4.png')
//     case 'age_5':
//     case 'badge_5':
//       return require('../../Assets/Images/Milestones/step5.png')
//   }
// }

// export default class ModalAchievementScreen extends Component {
//   state = {
//     animation: new Animated.Value(0)
//   }

//   static navigationOptions = {
//     header: null
//   }

//   setAnim = (anim) => {
//     if (!this.anim) {
//       this.anim = anim
//       this.anim.play()
//     }
//   }

//   onPressOk = () => {
//     if (this.props.achievementDisplayed) {
//       this.props.achievementDisplayed(this.props.achievement.key)
//     }
//   }

//   render () {
//     if (!this.props.achievement) {
//       return null
//     }
//     const {
//       key,
//       texts
//     } = this.props.achievement

//     if (!key || !texts) {
//       return null
//     }

//     const {
//       allCenter,
//       title1,
//       title2,
//       subtitle,
//       description
//     } = texts

//     const altLayout = texts.altLayout && !allCenter

//     return (
//       <Modal
//         onDismiss={this.onPressOk}
//         onRequestClose={this.onPressOk}
//         testID='AchievementScreen'
//       >
//         <LinearGradient
//           colors={['rgb(133, 155, 224)', 'rgb(82, 176, 208)']}
//           style={styles.container}
//           testID='AchievementScreen_Gradient'
//         >
//           {
//             // <LottieView
//             //   ref={this.setAnim}
//             //   source={altLayout ? Lighting2 : Lighting1}
//             //   resizeMode='cover'
//             //   contentMode='aspectFill'
//             //   style={StyleSheet.absoluteFill}
//             //   testID={altLayout ? 'AchievementScreen_LottieAlt' : 'AchievementScreen_Lottie'}
//             //   autoSize
//             // />
//           }
//           {!altLayout && (
//             <View
//               style={styles.top}
//               testID='AchievementScreen_Top'
//             >
//               <Text
//                 style={[styles.title, allCenter && styles.titleCenter]}
//                 testID={allCenter ? 'AchievementScreen_TopTitleCenter' : 'AchievementScreen_TopTitle'}
//               >{title1}</Text>
//               {(!!subtitle && !allCenter) && (
//                 <Text
//                   style={styles.subtitle}
//                   testID='AchievementScreen_TopSubtitle'
//                 >{subtitle}</Text>
//               )}
//             </View>
//           )}
//           <View
//             style={altLayout ? styles.imageRight : styles.imageCenter}
//             testID={altLayout ? 'AchievementScreen_AltImage' : 'AchievementScreen_CenterImage'}
//           >
//             {altLayout && (
//               <Text
//                 style={[styles.title, styles.titleWithImage]}
//                 testID='AchievementScreen_AltTitle'
//               >{title1}</Text>
//             )}
//             <Image
//               testID='AchievementScreen_Image'
//               source={getAchievementImage(key)}
//             />
//           </View>
//           {altLayout && (
//             <View
//               style={styles.bottom}
//               testID='AchievementScreen_Bottom'
//             >
//               <Text
//                 style={styles.title}
//                 testID='AchievementScreen_BottomTitle'
//               >{title2}</Text>
//               <Text
//                 style={styles.subtitle}
//                 testID='AchievementScreen_BottomSubtitle'
//               >{subtitle}</Text>
//             </View>
//           )}
//           <View
//             style={[styles.footer, allCenter && styles.footerCenter]}
//             testID={allCenter ? 'AchievementScreen_FooterCenter' : 'AchievementScreen_Footer'}
//           >
//             <Text
//               style={[styles.description, !allCenter && styles.descriptionFullWidth]}
//               testID={allCenter ? 'AchievementScreen_DescriptionCenter' : 'AchievementScreen_Description'}
//             >{description}</Text>
//             <RoundButton
//               minimal
//               onPress={this.onPressOk}
//               small
//               testID='AchievementScreen_Ok'
//               white
//             >Okay!</RoundButton>
//           </View>
//         </LinearGradient>
//       </Modal>
//     )
//   }
// }

// const styles = StyleSheet.createProportional({
//   container: {
//     flex: 1,
//     backgroundColor: 'transparent',
//     paddingVertical: 35
//   },
//   title: {
//     ...Fonts.Light,
//     fontSize: 50,
//     lineHeight: 60,
//     color: 'white',
//     paddingTop: 10
//   },
//   titleCenter: {
//     paddingTop: 10,
//     textAlign: 'center'
//   },
//   titleWithImage: {
//     position: 'absolute',
//     left: 40,
//     bottom: -5
//   },
//   subtitle: {
//     ...Fonts.SemiBold,
//     fontSize: 14,
//     color: 'white',
//     lineHeight: 20
//   },
//   top: {
//     paddingHorizontal: 40
//   },
//   bottom: {
//     paddingBottom: 20,
//     paddingHorizontal: 40
//   },
//   imageCenter: {
//     flex: 1,
//     justifyContent: 'center',
//     alignItems: 'center'
//   },
//   imageRight: {
//     flex: 1,
//     flexDirection: 'row',
//     justifyContent: 'flex-end',
//     alignItems: 'flex-end',
//     paddingBottom: 10
//   },
//   description: {
//     ...Fonts.Light,
//     fontSize: 12,
//     color: 'white',
//     marginRight: 18,
//     minHeight: 30
//   },
//   descriptionFullWidth: {
//     flex: 1,
//     minHeight: 38
//   },
//   footer: {
//     flexDirection: 'row',
//     alignItems: 'flex-end',
//     paddingHorizontal: 40
//   },
//   footerCenter: {
//     paddingTop: 15,
//     flexDirection: 'column',
//     alignItems: 'center',
//     justifyContent: 'center'
//   },
//   backgroundImage: {
//     width: Dimensions.get('screen').width,
//     height: Dimensions.get('screen').height
//   }
// })

// ModalAchievementScreen.propTypes = {
//   achievementDisplayed: PropTypes.func,
//   achievement: PropTypes.shape({
//     key: PropTypes.string,
//     texts: PropTypes.shape({
//       altLayout: PropTypes.bool,
//       allCenter: PropTypes.bool,
//       title1: PropTypes.string,
//       title2: PropTypes.string,
//       subtitle: PropTypes.string,
//       description: PropTypes.string
//     })
//   })
// }
