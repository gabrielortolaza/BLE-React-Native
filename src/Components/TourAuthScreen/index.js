import React, { Component } from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import { connect } from "react-redux";
import AsyncStorage from "@react-native-async-storage/async-storage";

import { forgotPassword, signWithEmail } from "../../Actions/Auth";
import { addMessage, setRequesting } from "../../Actions/Status";
import StyleSheet from "../../Proportional";
import { Colors, Fonts } from "../../Themes";
import {
  ButtonRound, InputField, Checkable,
  Label as Text, Loading
} from "../Shared";
import Container from "../Shared/Container";

class TourAuthScreen extends Component {
  constructor(props) {
    super(props);

    this.state = {
      displayName: "",
      email: "",
      password: "",
      termsAccepted: false,
      receiveNewsletters: false,
      routeName: props.route.params.routeName,
      rememberMeSelected: false
    };
  }

  componentDidMount() {
    this.getRememberMe();
  }

  getRememberMe = () => {
    const { routeName } = this.state;

    if (routeName === "SignIn") {
      AsyncStorage.getItem("rememberMeAuth").then((val) => {
        if (val) {
          this.setState({
            rememberMeSelected: true,
            email: val
          });
        }
      });
    } else {
      this.setState({
        email: ""
      });
    }
  };

  // Makes first letter in words capital
  toTitleCase = (string) => {
    return string
      .toLowerCase()
      .split(" ")
      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
      .join(" ");
  };

  signWithEmail = () => {
    const { isRequesting, setRequesting, signWithEmail } = this.props;
    const {
      displayName, email, password,
      termsAccepted, routeName, rememberMeSelected, receiveNewsletters
    } = this.state;
    console.log("isRequesting:", isRequesting);

    if (!isRequesting) {
      this.refInputDisplayName && this.refInputDisplayName.blur();
      this.refInputEmail && this.refInputEmail.blur();
      this.refInputPassword && this.refInputPassword.blur();

      const { navigation } = this.props;
      const displayName1 = this.toTitleCase(displayName);
      setRequesting("emailAuth");

      signWithEmail(
        {
          email,
          password,
          displayName: displayName1,
          termsAccepted,
          receiveNewsletters,
          routeName
        },
        navigation
      );

      this.setState({ password: "" });

      if (rememberMeSelected) {
        AsyncStorage.setItem("rememberMeAuth", email);
      } else {
        AsyncStorage.removeItem("rememberMeAuth");
      }
    }
  };

  forgotPassword = () => {
    const { isRequesting, forgotPassword, setRequesting } = this.props;

    if (!isRequesting) {
      this.refInputEmail && this.refInputEmail.blur();
      const { email } = this.state;

      setRequesting("emailAuth");

      forgotPassword({ email });
    }
  };

  goBack = () => {
    const { navigation, setRequesting } = this.props;
    setRequesting("");
    navigation.goBack();
  };

  handleOnPressRadio = () => {
    const { termsAccepted } = this.state;
    this.setState({ termsAccepted: !termsAccepted });
  };

  onInputDisplayNameRef = (refInputDisplayName) => {
    this.refInputDisplayName = refInputDisplayName;
  };

  onInputEmailRef = (refInputEmail) => {
    this.refInputEmail = refInputEmail;
  };

  onInputPasswordRef = (refInputPassword) => {
    this.refInputPassword = refInputPassword;
  };

  focusEmail = () => {
    this.refInputEmail.focus();
  };

  focusPassword = () => {
    this.refInputPassword.focus();
  };

  setDisplayName = (displayName) => this.setState({ displayName });

  setEmail = (email) => this.setState({ email: email.trim() });

  setPassword = (password) => this.setState({ password });

  emptyFn = () => {};

  renderIndicator = () => null;

  render() {
    const { navigation, isRequesting } = this.props;
    const {
      termsAccepted, routeName, rememberMeSelected,
      email, password, receiveNewsletters
    } = this.state;

    const isSignIn = routeName === "SignIn";
    const isSignUp = !isSignIn;

    return (
      <Container edges={["top"]} testID="TourAuthScreen" style={styles.container}>
        <>
          <View testID="TourAuth_Welcome" style={styles.header}>
            <View style={styles.headerTextView}>
              <Text grey font16 weightSemiBold style={styles.headerText}>
                {isSignIn ? "Log in" : "Sign up"}
              </Text>
            </View>
          </View>
          <View style={styles.introText}>
            <Text font16 grey>
              {
                isSignIn ? "Good to see you again!" : "Let's get started!"
              }
            </Text>
          </View>
          <View style={styles.content} testID="TourAuth_Inputs">
            {isSignUp && (
              <InputField
                testID="TourAuth_Name_TextInput"
                autoCapitalize="words"
                autoCorrect={false}
                onChangeText={this.setDisplayName}
                onSubmitEditing={this.focusEmail}
                placeholder="Name"
                ref={this.onInputDisplayNameRef}
                returnKeyType="next"
                style={styles.input}
              />
            )}
            <InputField
              testID="TourAuth_Email_TextInput"
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="email-address"
              onChangeText={this.setEmail}
              value={email}
              onSubmitEditing={this.focusPassword}
              placeholder="Email"
              ref={this.onInputEmailRef}
              returnKeyType="next"
              style={styles.input}
            />
            <InputField
              testID="TourAuth_Password_TextInput"
              value={password}
              autoCapitalize="none"
              autoCorrect={false}
              onChangeText={this.setPassword}
              onSubmitEditing={this.sign}
              placeholder="Password"
              returnKeyType="done"
              ref={this.onInputPasswordRef}
              secureTextEntry
              style={styles.input}
            />
            {isSignUp && (
              <View style={styles.radioButtonView}>
                <Checkable
                  testID="TourAuth_Terms"
                  onPress={this.handleOnPressRadio}
                  selected={termsAccepted}
                  style={styles.checkable}
                >
                  <Text onPress={() => navigation.navigate("Terms")} style={styles.terms}>
                    <Text>I agree to the Pumpables</Text>
                    <Text style={styles.termsBold}> Terms of Service </Text>
                    <Text>and</Text>
                    <Text style={styles.termsBold}> Privacy Policy</Text>
                  </Text>
                </Checkable>
                <Checkable
                  testID="TourAuth_Newsletters"
                  onPress={() => this.setState({ receiveNewsletters: !receiveNewsletters })}
                  selected={receiveNewsletters}
                >
                  <Text style={styles.terms}>
                    Receive marketing info
                  </Text>
                </Checkable>
              </View>
            )}
            {isSignIn && (
              <View>
                <Checkable
                  testID="remember-me-auth"
                  onPress={() => { this.setState({ rememberMeSelected: !rememberMeSelected }); }}
                  selected={rememberMeSelected}
                  style={styles.checkable}
                >
                  <Text font12>Remember me</Text>
                </Checkable>
                <ButtonRound
                  onPress={this.forgotPassword}
                  style={styles.forgotPassword}
                  transparent
                  testID="TourAuthSelection_Forgot"
                >
                  <Text lightGrey2 style={styles.forgotPasswordText}>I forgot my password</Text>
                </ButtonRound>
              </View>
            )}
            <ButtonRound
              testID="TourAuth_Start"
              onPress={this.signWithEmail}
              blue
              style={[styles.button, { marginTop: 50 }]}
            >
              <Text white weightSemiBold font20>{isSignUp ? "Start Now" : "Log in"}</Text>
            </ButtonRound>
            <ButtonRound
              onPress={() => {
                this.setState({ routeName: isSignIn ? "SignUp" : "SignIn" });
                setTimeout(() => { this.getRememberMe(); }, 250);
              }}
              style={styles.button}
              transparent
              testID="TourAuthSelection_Already_Have_Account?"
            >
              <Text lightGrey2 font14>
                <Text>
                  {isSignIn ? "Don't " : "Already "}
                  have an account?
                </Text>
                <Text weightBold>
                  {isSignIn ? " Sign up" : " Log in"}
                </Text>
              </Text>
            </ButtonRound>
          </View>
        </>
        <Loading isLoading={isRequesting} />
      </Container>
    );
  }
}

TourAuthScreen.propTypes = {
  forgotPassword: PropTypes.func.isRequired,
  isRequesting: PropTypes.bool,
  route: PropTypes.object,
  navigation: PropTypes.object.isRequired,
  signWithEmail: PropTypes.func.isRequired,
  setRequesting: PropTypes.func
};

const styles = StyleSheet.createProportional({
  container: {
    flex: 1,
    backgroundColor: Colors.white
  },
  content: {
    marginHorizontal: 25,
    marginTop: 20,
    flex: 5
  },
  header: {
    flexDirection: "row",
    alignItems: "center",
    marginTop: 30
  },
  headerTextView: {
    marginLeft: 25
  },
  headerText: {
    fontSize: 40
  },
  goBack: {
    marginLeft: 19
  },
  goBackIcon: {
    color: Colors.black
  },
  introText: {
    marginLeft: 25,
    marginTop: 30
  },
  input: {
    marginBottom: 9,
    fontSize: 15
  },
  backButton: {
    position: "absolute",
    top: 30,
    padding: 10,
    left: 20,
    zIndex: 10
  },
  button: {
    alignSelf: "center",
    width: "100%",
    marginTop: 15
  },
  radioButtonView: {
    marginTop: 15
  },
  radioBackground: {
    width: 20,
    height: 20,
    // borderRadius: 10,
    padding: 6,
    borderWidth: 1,
    borderColor: Colors.lightBlue,
    backgroundColor: Colors.backgroundTwo,
    justifyContent: "center",
    alignItems: "center"
  },
  radioDot: {
    width: 8,
    height: 8,
    borderRadius: 4,
    backgroundColor: "transparent"
  },
  radioDotSelected: {
    width: 8,
    height: 8,
    borderRadius: 4,
    backgroundColor: Colors.windowsBlue
  },
  terms: {
    color: Colors.grey,
    fontSize: 13,
    lineHeight: 20
  },
  indicatorWrapper: {
    height: 36
  },
  termsBold: {
    ...Fonts.SemiBold
  },
  errorMessage: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    height: 69,
    backgroundColor: Colors.coral,
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 50
  },
  errorText: {
    color: Colors.white,
    fontSize: 14,
    fontWeight: "bold",
    textAlign: "center"
  },
  checkable: {
    marginVertical: 15,
    paddingRight: 25
  },
  forgotPassword: {
    width: 220,
    justifyContent: "flex-end",
    alignSelf: "flex-end",
    paddingRight: 0,
    marginTop: 15
  },
  forgotPasswordText: {
    fontSize: 13,
    textDecorationLine: "underline"
  }
});

const mapStateToProps = ({ status }) => ({
  isRequesting: status.requesting === "emailAuth"
});

const mapDispatchToProps = {
  forgotPassword,
  addMessage,
  signWithEmail,
  setRequesting
};

export default connect(mapStateToProps, mapDispatchToProps)(TourAuthScreen);
