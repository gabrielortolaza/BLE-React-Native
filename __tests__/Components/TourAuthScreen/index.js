import React from "react";
import PropTypes from "prop-types";
import { shallow } from "enzyme";

import { TourAuthScreen } from "../../../src/Components/TourAuthScreen";

const USER_DISPLAY_NAME = "A Nerdy Factorer";
const USER_DISPLAY_EMAIL = "nkvdamian+ahdfbkjfn@gmail.com";
const USER_DISPLAY_PASSWORD = "123456789";

describe("TourAuthScreen", () => {
  // it("Sign Up - with empty inputs", () => {
  //   const goBack = jest.fn();
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={goBack}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   expect(render.find({ testID: "TourAuth_Welcome" })).toHaveLength(1);
  //   expect(render.find({ testID: "TourAuth_Back" })).toHaveLength(1);
  //   expect(render.find({ testID: "TourAuth_Name_TextInput" })).toHaveLength(1);
  //   expect(render.find({ testID: "TourAuth_Email_TextInput" })).toHaveLength(1);
  //   expect(render.find({ testID: "TourAuth_Password_TextInput" })).toHaveLength(1);
  //   expect(render.find({ testID: "TourAuth_Terms" })).toHaveLength(1);
  //
  //   expect(goBack.mock.calls).toHaveLength(0);
  //   render.find({ testID: "TourAuth_Back" }).simulate("press");
  //   expect(goBack.mock.calls).toHaveLength(1);
  //
  //   expect(signWithEmail.mock.calls).toHaveLength(0);
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   expect(signWithEmail.mock.calls).toHaveLength(1);
  //
  //   const args = signWithEmail.mock.calls[0][0]; // First argument is an object
  //   expect(args.displayName).toBeFalsy();
  //   expect(args.email).toBeFalsy();
  //   expect(args.password).toBeFalsy();
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Name", () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Name_TextInput" }).simulate("changeText", USER_DISPLAY_NAME);
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   expect(signWithEmail.mock.calls[0][0].displayName).toBe(USER_DISPLAY_NAME);
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Email", () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Email_TextInput" }).simulate("changeText", USER_DISPLAY_EMAIL);
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   expect(signWithEmail.mock.calls[0][0].email).toBe(USER_DISPLAY_EMAIL);
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Name + Email", () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Name_TextInput" }).simulate("changeText", USER_DISPLAY_NAME);
  //   render.find({ testID: "TourAuth_Email_TextInput" }).simulate("changeText", USER_DISPLAY_EMAIL);
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //
  //   const args = signWithEmail.mock.calls[0][0]; // First argument is an object
  //   expect(args.displayName).toBe(USER_DISPLAY_NAME);
  //   expect(args.email).toBe(USER_DISPLAY_EMAIL);
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Password", async () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Password_TextInput" }).simulate("changeText", USER_DISPLAY_PASSWORD);
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   await new Promise((resolve) => resolve(), 200);
  //   expect(signWithEmail.mock.calls[0][0].password).toBe(USER_DISPLAY_PASSWORD);
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Name + Email + Password", async () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Name_TextInput" }).simulate("changeText", USER_DISPLAY_NAME);
  //   render.find({ testID: "TourAuth_Email_TextInput" }).simulate("changeText", USER_DISPLAY_EMAIL);
  //   render.find({ testID: "TourAuth_Password_TextInput" }).simulate("changeText", USER_DISPLAY_PASSWORD);
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   await new Promise((resolve) => resolve(), 200);
  //
  //   const args = signWithEmail.mock.calls[0][0]; // First argument is an object
  //   expect(args.displayName).toBe(USER_DISPLAY_NAME);
  //   expect(args.email).toBe(USER_DISPLAY_EMAIL);
  //   expect(args.password).toBe(USER_DISPLAY_PASSWORD);
  //
  //   expect(render).toMatchSnapshot();
  // });
  //
  // it("Sign Up - with Name + Email + Password + Terms", async () => {
  //   const signWithEmail = jest.fn();
  //
  //   const render = shallow(
  //     <TourAuthScreen
  //       goBack={jest.fn()}
  //       navigation={{ state: { routeName: "SignUp" } }}
  //       signWithEmail={signWithEmail}
  //     />
  //   ).dive();
  //
  //   render.find({ testID: "TourAuth_Name_TextInput" }).simulate("changeText", USER_DISPLAY_NAME);
  //   render.find({ testID: "TourAuth_Email_TextInput" }).simulate("changeText", USER_DISPLAY_EMAIL);
  //   render.find({ testID: "TourAuth_Password_TextInput" }).simulate("changeText", USER_DISPLAY_PASSWORD);
  //
  //   render.find({ testID: "TourAuth_Terms" }).simulate("press");
  //
  //   render.find({ testID: "TourAuth_Start" }).simulate("press");
  //   await new Promise((resolve) => resolve(), 200);
  //
  //   const args = signWithEmail.mock.calls[0][0]; // First argument is an object
  //   expect(args.displayName).toBe(USER_DISPLAY_NAME);
  //   expect(args.email).toBe(USER_DISPLAY_EMAIL);
  //   expect(args.password).toBe(USER_DISPLAY_PASSWORD);
  //   expect(args.termsAccepted).toBe(true);
  //
  //   expect(render).toMatchSnapshot();
  // });

  it("Sign In - with empty inputs", () => {
    const signWithEmail = jest.fn();

    const render = shallow(
      <TourAuthScreen
        setRequesting={jest.fn()}
        navigation={{ goBack: jest.fn() }}
        signWithEmail={signWithEmail}
      />
    ).dive();

    expect(render.find({ testID: "TourAuth_Name_TextInput" })).toHaveLength(0);
    expect(render.find({ testID: "TourAuth_Terms" })).toHaveLength(0);

    render.find({ testID: "TourAuth_Start" }).simulate("press");

    const args = signWithEmail.mock.calls[0][0]; // First argument is an object
    console.log("Args:", args);
    expect(args.displayName).toBeFalsy();
    expect(args.email).toBeFalsy();
    expect(args.password).toBeFalsy();

    expect(render).toMatchSnapshot();
  });

  it("Sign In - with email and password", () => {
    const signWithEmail = jest.fn();

    const render = shallow(
      <TourAuthScreen
        setRequesting={jest.fn()}
        navigation={{ goBack: jest.fn() }}
        signWithEmail={signWithEmail}
      />
    ).dive();

    expect(render.find({ testID: "TourAuth_Name_TextInput" })).toHaveLength(0);
    expect(render.find({ testID: "TourAuth_Terms" })).toHaveLength(0);

    render.find({ testID: "TourAuth_Email_TextInput" }).simulate("changeText", USER_DISPLAY_EMAIL);
    render.find({ testID: "TourAuth_Password_TextInput" }).simulate("changeText", USER_DISPLAY_PASSWORD);

    render.find({ testID: "TourAuth_Start" }).simulate("press");

    const args = signWithEmail.mock.calls[0][0]; // First argument is an object
    expect(args.displayName).toBeFalsy();
    expect(args.email).toBe(USER_DISPLAY_EMAIL);
    expect(args.password).toBe(USER_DISPLAY_PASSWORD);

    expect(render).toMatchSnapshot();
  });
});

TourAuthScreen.propTypes = {
  goBack: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired,
  signWithEmail: PropTypes.func.isRequired
};
