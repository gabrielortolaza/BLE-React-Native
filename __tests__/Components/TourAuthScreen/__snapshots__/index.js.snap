// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourAuthScreen Sign In - with email and password 1`] = `ShallowWrapper {}`;

exports[`TourAuthScreen Sign In - with empty inputs 1`] = `ShallowWrapper {}`;
